openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -874,13 +870,32 @@ port_INLINE void activity_ti1ORri1() {
       // find the next one
       ieee154e_vars.nextActiveSlotOffset = schedule_getNextActiveSlotOffset();
    } else {
-      // this is NOT the next active slot, abort
+      // this is NOT the next active slot
       // stop using serial
       openserial_stop();
-      // abort the slot
-      endSlot();
-      //start outputing serial
-      openserial_startOutput();
+      // check if I can send an EB
+      ieee154e_vars.dataToSend = openqueue_macGetEBPacket();
+      if (ieee154e_vars.dataToSend != NULL) {
+         ieee154e_vars.isSpouriousEB = TRUE;
+         // change state
+         changeState(S_TXDATAOFFSET);
+         // change owner
+         ieee154e_vars.dataToSend->owner = COMPONENT_IEEE802154E;

```

Comment:
Please move the following lines to a function which gets called here, and in the place to copy-pasted this code from.